### PR TITLE
Make clean command top level

### LINKED ISSUE #10783: Not clear how to "clean" my flutter project, and remove all generated/built artifacts
I tried to run `flutter clean` but that didn't work. I looked at `flutter --help` and I didn't see anything with the semantics of "clean up all the generated/built artifacts". Maybe I missed it?

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/10783

/cc @sethladd

### CODE REVIEW

**Comment by sethladd** [comment]
YES

**Comment by chinmaygarde** [comment]
This wont clean up platform specific build system intermediates. For example,

```
android/app/src/main/java/*
android/gen/
android/local.properties
ios/Flutter/App.framework/
ios/Flutter/Flutter.framework/
ios/Flutter/Generated.xcconfig
ios/Flutter/app.flx
ios/Frameworks/
ios/Runner/GeneratedPluginRegistrant.h
ios/Runner/GeneratedPluginRegistrant.m
ios/ServiceDefinitions.json
```

**Review by tvolkert — APPROVED** [review]
LGTM

@chinmaygarde, that's true, but it's also true of the current `flutter build clean`.  I filed https://github.com/flutter/flutter/issues/12631 to track.

File: packages/flutter_tools/lib/src/commands/build.dart [hunk 1]
```diff
@@ -6,24 +6,23 @@ import 'dart:async';
...
+import 'clean.dart';
 
 class BuildCommand extends FlutterCommand {
   BuildCommand({bool verboseHelp: false}) {
     addSubcommand(new BuildApkCommand());
     addSubcommand(new BuildAotCommand());
-    addSubcommand(new BuildCleanCommand());
     addSubcommand(new BuildIOSCommand());
     addSubcommand(new BuildFlxCommand(verboseHelp: verboseHelp));
+    addSubcommand(new CleanCommand());  // here for backwards compatibility.
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/flutter_tools/lib/src/commands/build.dart]: OK, removed.

**Pull request merged**